pandas/core/strings.py

```diff
@@ -1095,24 +1095,48 @@ def str_pad(arr, width, side='left', fillchar=' '):
 
 def str_split(arr, pat=None, n=None):
     """
+    Split strings around given separator/delimiter.
+
     Split each string (a la re.split) in the Series/Index by given
     pattern, propagating NA values. Equivalent to :meth:`str.split`.
 
     Parameters
     ----------
     pat : string, default None
-        String or regular expression to split on. If None, splits on whitespace
+        String or regular expression to split on. If None, split on whitespace.
     n : int, default -1 (all)
-        None, 0 and -1 will be interpreted as return all splits
+        * None, 0 and -1 will be interpreted as return all splits.
+        * Vary output dimensionality if `expand` is True:
+            - If n >= default splits, return all splits.
+            - If n < default splits, makes first n splits only.
     expand : bool, default False
-        * If True, return DataFrame/MultiIndex expanding dimensionality.
+        * If True, return DataFrame/MultiIndex expanding dimensionality.\
+         Appends None for padding.
         * If False, return Series/Index.
 
-    return_type : deprecated, use `expand`
-
     Returns
     -------
     split : Series/Index or DataFrame/MultiIndex of objects
```

Comment:
I find this return type rather confusing - how does the Index play a part? I think this could be clarified better

Reply:
Yes, I was confused about that as well. Should I keep it in the doc?

Reply:
For returns make sure the first line is just the type (unless returning more than one, which isn't the case here). I think that would best be `Series or Index` with a description that says something like "Return type matches caller unless `expand=True` (always returns a `DataFrame`)"